service/create/pki.go

```diff
@@ -0,0 +1,98 @@
+package create
+
+import (
+	"fmt"
+	"strings"
+
+	microerror "github.com/giantswarm/microkit/error"
+
+	"github.com/giantswarm/certctl/service/pki"
+	"github.com/giantswarm/certctl/service/token"
+)
+
+// Creates a PKI backend if one does not exist for the cluster.
+func (s *Service) setupPKIBackend(cert CertificateSpec) error {
+	var err error
+
+	service, err := s.getPKIService()
+	if err != nil {
+		return microerror.MaskAny(err)
+	}
+
+	isValid, err := service.VerifyPKISetup(cert.ClusterID)
+	if err != nil {
+		return microerror.MaskAny(err)
+	}
+	if isValid {
+		s.Config.Logger.Log("debug", fmt.Sprintf("PKI backend already exists for cluster %s", cert.ClusterID))
+		return nil
+	} else {
+		caCommonName := s.Config.getCACommonName(cert)
+
+		// Create PKI backend
+		config := pki.CreateConfig{
+			ClusterID:        cert.ClusterID,
+			CommonName:       caCommonName,
+			AllowedDomains:   getAllowedDomainsForCA(caCommonName, cert),
+			AllowBareDomains: cert.AllowBareDomains,
+			TTL:              s.Config.Viper.GetString(s.Config.Flag.Vault.PKI.CATTL),
+		}
+
+		s.Config.Logger.Log("debug", fmt.Sprintf("Creating PKI backend for cluster %s", cert.ClusterID))
+		return service.Create(config)
```

Comment:
I find it weird that a private method calls a public one. Like, it should be the other way around. When I talked with Timo it turned out we want to have a CA and a cert operator at the end. So we should already start having different service packages in here so we can more easily factor them out later. Getting the design more right now is quite important I guess. So I suggest we have a `service/ca` service package which provides a `Create` method and all that stuff and then we have a `service/crt` service package with its own `Create` and other stuff. For now when having the cert-operator manage CAs and Certs, the `service/crt` service should import and use the `service/ca` service. Later we can decouple them more easily. WDYT?